broker/lua/src/macro_cache.cc

```diff
@@ -809,14 +819,36 @@ void macro_cache::_process_pb_adaptive_host(
  *  @param data  The event.
  */
 void macro_cache::_process_host_group(std::shared_ptr<io::data> const& data) {
-  std::shared_ptr<neb::host_group> const& hg =
+  const std::shared_ptr<neb::host_group>& hg =
       std::static_pointer_cast<neb::host_group>(data);
   SPDLOG_LOGGER_DEBUG(log_v2::lua(),
                       "lua: processing host group '{}' of id {} enabled: {}",
                       hg->name, hg->id, hg->enabled);
-  if (hg->enabled)
-    _host_groups[hg->id] = hg;
-  // erasure is desactivated because a group cen be owned by several pollers
+  if (hg->enabled) {
+    auto found = _host_groups.find(hg->id);
+    if (found != _host_groups.end()) {
+      /* here, we complete the set of pollers */
+      found->second.second.insert(hg->poller_id);
+      found->second.first->name = hg->name;
+    } else {
+      /* Here, we add the hostgroup and the first poller that needs it */
+      absl::flat_hash_set<uint32_t> pollers{hg->poller_id};
+      _host_groups[hg->id] = std::make_pair(std::move(hg), pollers);
```

Comment:
Since hg is already a reference, I think you should remove the std::move

Reply:
will fix